lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -586,6 +599,63 @@ class QcFlagService {
         }));
     }
 
+    /**
+     * Get GAQ summary
+     *
+     * @param {number} dataPassId id of data pass id
+     * @return {Promise<GaqSummary[]>} Resolves with the GAQ Summary
+     */
+    async getGaqSummary(dataPassId) {
+        const runGaqSubSummaries = await QcFlagRepository.getRunGaqSubSummaries(dataPassId);
+
+        const summary = {};
+        const flagsAndVerifications = {};
+
+        // Fold list of aggregations into summary
+        for (const subSummary of runGaqSubSummaries) {
+            const {
+                runNumber,
+                bad,
+                effectiveRunCoverage,
+                flagsIds,
+                verifiedFlagsIds,
+                mcReproducible,
+            } = subSummary;
+
+            if (!summary[runNumber]) {
+                summary[runNumber] = { [QC_SUMMARY_PROPERTIES.mcReproducible]: false };
+            }
+            if (!flagsAndVerifications[runNumber]) {
+                flagsAndVerifications[runNumber] = {};
+            }
+
+            const runSummary = summary[runNumber];
+            const runFlagsAndVerifications = flagsAndVerifications[runNumber];
+
+            runFlagsAndVerifications.flagsIds = new Set([...runFlagsAndVerifications?.flagsIds ?? [], ...flagsIds]);
```

Comment:
You create a set out of the flags, but then in the `for loop` on line 652 you still use the initial array rather than the set. 
Is this intended?

Reply:
Ok, you use it via reference but this makes it more difficult to read. Why not something along the lines:
```suggestion
            const runFlagsIds = flagsAndVerifications[runNumber]?.flagsIds ?? [];
            const runVerifiedFlagsIds = flagsAndVerifications[runNumber]?.verifiedFlagsIds ?? [];

            flagsAndVerifications[runNumber] = {
                flagsIds: new Set([...runFlagsIds, ...flagsIds]),
                verifiedFlagsIds: new Set([ runVerifiedFlagsIds, ...verifiedFlagsIds])
            };
```